### PR TITLE
Split project edition in separate tabs

### LINKED ISSUE #31: Split “Project settings” page into tabs
The “Project settings” page is quite visually heavy at the moment, splitting it in tabs (like in the audits page) will make it more easy to navigate.

We can split it into 4 tabs:

- General
- Project environments
- Pages & scripts
- Members

This will also help streamlining the project creation flow (4 short steps instead of a single big one).

### PR DESCRIPTION
**Pull request by LouisPinsard**
Resolves #31 

image
image
image
image

### CODE REVIEW

File: frontend/src/pages/GeneralSettings/GeneralSettings.style.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    color: ${colorUsage.h1Text};
+    font-family: ${fontFamily.mainSans};
+    font-size: ${fontSize.h1Text};
+    font-weight: ${fontWeight.h1Text};
+    text-align: center;
+    overflow: hidden;
+    text-overflow: ellipsis;
+  `,
+
+Title: styled.div`
```
Comment (phacks) [inline frontend/src/pages/GeneralSettings/GeneralSettings.style.tsx]: `h1` or `h2` no?

File: frontend/src/pages/GeneralSettings/GeneralSettings.style.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  line-height: ${lineHeight.h2Text};
+  color: ${colorUsage.h2Text};
+  font-family: ${fontFamily.mainSans};
+  font-size: ${fontSize.h2Text};
+  font-weight: ${fontWeight.h2Text};
+  margin-top: ${getSpacing(6)};
+`,
+
+PageSubTitle: styled.div`
```
Comment (phacks) [inline frontend/src/pages/GeneralSettings/GeneralSettings.style.tsx]: `h2` or `h3`

File: frontend/src/pages/GeneralSettings/GeneralSettings.style.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+`,
+
+SelectUser: styled(Select)`
+  margin-top: ${getSpacing(4)};
+  font-family: ${fontFamily.mainSans};
+  font-size: ${fontSize.inputSelectText};
+  line-height: ${lineHeight.inputSelectText};
+`,
+
+InviteUserLink: styled.div`
```
Comment (phacks) [inline frontend/src/pages/GeneralSettings/GeneralSettings.style.tsx]: `p` or `span`?

File: frontend/src/components/Root/components/Menu/components/ProjectSettingsMenuContent/ProjectSettingsMenuContent.style.ts [hunk 4]
```diff
@@ -12,3 +16,64 @@ export const GoBackToProjectLink = styled.a`
...
+  line-height: ${lineHeight.h3Text};
+  color: ${colorUsage.h3Text};
+  font-family: ${fontFamily.mainSans};
+  font-size: ${fontSize.h3Text};
+  font-weight: ${fontWeight.h3Text};
+  padding: 0 ${getSpacing(3)} ${getSpacing(4)} ${getSpacing(3)};
+  margin-top: ${getSpacing(4)}
+`;
+
+export const SettingsPageTitleBlock = styled.div`
```
Comment (phacks) [inline frontend/src/components/Root/components/Menu/components/ProjectSettingsMenuContent/ProjectSettingsMenuContent.style.ts]: same here (and I guess on other titles)

File: frontend/src/components/Root/components/Menu/components/ProjectSettingsMenuContent/ProjectSettingsMenuContent.tsx [hunk 5]
```diff
@@ -3,28 +3,75 @@ import React from 'react';
...
+  currentURL: string;
+} & OwnProps & InjectedIntlProps;
 
 export const ProjectSettingsMenuContent: React.FunctionComponent<Props> = ({
   project,
+  currentURL,
+  intl,
 }) => {
 
+  const settingPages = [
```
Comment (phacks) [inline frontend/src/components/Root/components/Menu/components/ProjectSettingsMenuContent/ProjectSettingsMenuContent.tsx]: oh, clever!

File: frontend/src/components/Root/components/Menu/components/ProjectSettingsMenuContent/ProjectSettingsMenuContent.tsx [hunk 6]
```diff
@@ -3,28 +3,75 @@ import React from 'react';
...
 
   return (
-    <GoBackToProjectLink onClick={redirectToProjectPage} >
-      <FormattedMessage id="Menu.go_back_to_project" />
-    </GoBackToProjectLink>
+    <Container>
+      <GoBackToProjectLink onClick={redirectToProjectPage} >
+        <FormattedMessage id="Menu.go_back_to_project" />
+      </GoBackToProjectLink>
+      <Settings>Settings</Settings>
```
Comment (phacks) [inline frontend/src/components/Root/components/Menu/components/ProjectSettingsMenuContent/ProjectSettingsMenuContent.tsx]: This should be translated I guess

File: frontend/src/pages/EnvironmentSettings/EnvironmentSettings.style.tsx [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+import styled from 'styled-components';
+import { colorUsage, fontFamily, fontSize, fontWeight, getSpacing, lineHeight } from 'stylesheet';
+
+const Style = {
+  Container: styled.div`
+    display: flex;
+    flex-direction: column;
+    padding: 0 0 ${getSpacing(4)} 0;
+    margin: 0;
+    width: 800px;
```
Comment (phacks) [inline frontend/src/pages/EnvironmentSettings/EnvironmentSettings.style.tsx]: Is that `800px` value is used for all settings pages? could it be put inside the `stylesheet`?

File: frontend/src/components/Root/components/Menu/components/ProjectSettingsMenuContent/ProjectSettingsMenuContent.tsx [hunk 8]
```diff
@@ -31,7 +31,7 @@ export const ProjectSettingsMenuContent: React.FunctionComponent<Props> = ({
 }) => {
 
   const settingPages = [
-    routeDefinitions.projectSettingsGeneral, routeDefinitions.environmentSettings
+    routeDefinitions.projectSettingsGeneral, routeDefinitions.environmentSettings, routeDefinitions.scriptsAndPagesSettings
```
Comment (phacks) [inline frontend/src/components/Root/components/Menu/components/ProjectSettingsMenuContent/ProjectSettingsMenuContent.tsx]: Could you change to `pagesAndScripts` (even in the tab title)? Pages are 70%+ of all the audits on Falco, it makes sense to put it first 👍

File: frontend/src/pages/ScriptsAndPagesSettings/Components/ScriptTable/ScriptModal/ScriptModal.style.ts [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+  position: absolute;
+  top: 40px;
+  right: 10px;
+  width: 30px;
+  height: 30px;
+  cursor: pointer;
+  z-index: ${zIndex.graphModalCloseButton};
+`;
+
+export const PageTitle = styled.span`
```
Comment (phacks) [inline frontend/src/pages/ScriptsAndPagesSettings/Components/ScriptTable/ScriptModal/ScriptModal.style.ts]: title => `h*` to be more semantic

File: frontend/src/pages/ScriptsAndPagesSettings/Components/ScriptTable/ScriptModal/ScriptModal.style.ts [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+  background-color: ${colorUsage.createScriptButtonBackground};
+  font-family: ${fontFamily.mainSans};
+  font-size: ${fontSize.createScriptButtonText};
+  font-weight: bold;
+  line-height: ${lineHeight.createScriptButtonText};
+  cursor: pointer;
+  text-decoration: none;
+`;
+
+export const Loader = styled.div`
```
Comment (phacks) [inline frontend/src/pages/ScriptsAndPagesSettings/Components/ScriptTable/ScriptModal/ScriptModal.style.ts]: Could we reuse the existing `Loader` component instead?

**Comment by phacks** [comment]
Could we keep the settings page title (currently the project name) on all settings page, and maybe changing it to `Project Settings — <Project name>` (`Paramètres du projet — Theodo`)? e.g.

image

**Comment by phacks** [comment]
I’m wondering, now that the settings pages are nearly complete, what we could refactor. It looks like some logic (and a lot of styling) could be factorized. We can have a look at that in another PR though!

**Comment by LouisPinsard** [comment]
@phacks I chose the first option for the page title ! It's a bit complicated to reuse the existing loader, I would have to add a lot of new props to the component. I fixed your other remarks

**Comment by phacks** [comment]
@LouisPinsard great, thanks!

> It's a bit complicated to reuse the existing loader, I would have to add a lot of new props to the component.

In the new version of the PR it looks like you managed to reuse `components/Loader` though? Can’t find the `export const Loader` that I saw yesterday

Could you create an issue for refactoring the components in the settings page also?

**Comment by LouisPinsard** [comment]
The `export const Loader` is still here, I will create an issue to refactor components !

**Pull request merged**